src/porepy/applications/verification_setups/manu_poromech_nofrac.py

```diff
@@ -652,94 +796,64 @@ def after_nonlinear_convergence(
         darcy_flux_ad = self.darcy_flux([sd])
         darcy_flux_vals = darcy_flux_ad.evaluate(self.equation_system).val
-        data[pp.STATE][pp.ITERATE]["darcy_flux"] = darcy_flux_vals
+        data[pp.STATE][pp.ITERATE][self.flux_variable] = darcy_flux_vals
 
-        # Recover poroelastic force and store it in pp.ITERATE
+        # Recover poroelastic forces and store them in pp.ITERATE
         force_ad = self.stress([sd])
         force_vals = force_ad.evaluate(self.equation_system).val
-        data[pp.STATE][pp.ITERATE]["poroelastic_force"] = force_vals
+        data[pp.STATE][pp.ITERATE][self.stress_variable] = force_vals
 
-        # Store results
-        t = self.time_manager.time
-        tf = self.time_manager.time_final
-        if np.any(np.isclose(t, self.params.get("stored_times", [tf]))):
-            self.results.append(StoreResults(self))
+        # Inherit from base class
+        super().after_nonlinear_convergence(solution, errors, iteration_counter)
 
     def after_simulation(self) -> None:
         """Method to be called after the simulation has finished."""
         if self.params.get("plot_results", False):
             self.plot_results()
 
-    # -----> Plotting-related methods
-    def plot_results(self) -> None:
-        """Plotting results"""
-        self._plot_displacement()
-        self._plot_pressure()
-
-    def _plot_pressure(self):
-        """Plot exact and numerical pressures"""
-        sd = self.mdg.subdomains()[0]
-        p_ex = self.results[-1].exact_pressure
-        p_num = self.results[-1].approx_pressure
-        pp.plot_grid(sd, p_ex, plot_2d=True, linewidth=0, title="pressure (Exact)")
-        pp.plot_grid(sd, p_num, plot_2d=True, linewidth=0, title="pressure (Numerical)")
-
-    def _plot_displacement(self):
-        """Plot exact and numerical displacements."""
-        sd = self.mdg.subdomains()[0]
-        u_ex = self.results[-1].exact_displacement
-        u_num = self.results[-1].approx_displacement
-        # Horizontal displacement
-        pp.plot_grid(sd, u_ex[::2], plot_2d=True, linewidth=0, title="u_x (Exact)")
-        pp.plot_grid(sd, u_num[::2], plot_2d=True, linewidth=0, title="u_x (Numerical)")
-        # Vertical displacement
-        pp.plot_grid(sd, u_ex[1::2], plot_2d=True, linewidth=0, title="u_y (Exact)")
-        pp.plot_grid(
-            sd, u_num[1::2], plot_2d=True, linewidth=0, title="u_y (Numerical)"
-        )
-
 
-# ---------> Mixer class
-class ManuPoromechanics2d(  # type: ignore
-    UnitSquare,
+class ManufacturedNonlinearPoromechanicsNoFrac2d(  # type: ignore[misc]
+    UnitSquareTriangleGrid,
     ModifiedEquationsPoromechanics,
     ModifiedSolutionStrategy,
     poromechanics.Poromechanics,
+    SetupUtilities,
+    VerificationUtils,
+    ModifiedDataSavinMixin,
 ):
     """
-    Mixer class for the verification setup.
+    Mixer class for the two-dimensional non-linear poromechanics verification setup.
```

Comment:
Perhaps it suffices to point to the verification test file for usage example? Advantage is reduced maintenance (this example will be outdated at some point without us noticing).

Reply:
I agree, although, from the user side, it is extremely convenient to have something that can be pasted into another file and run without having to inspect a test. For example, Mandel's verification setup will, by default, replicate the results of the PorePy paper (this of course is not feasible for a light-weighted test), which has an added value in itself.

My vote goes for keeping the Examples section, even though it adds a slight maintenance overhead.

Reply:
Let's allow the others to weigh in on this one. Opinions, @keileg, @OmarDuran?

Reply:
I will need a short briefing before making an opinion.

Reply:
I think I see the question now. While I agree with @jhabriel that such code snippets can be extremely useful, I also share @IvarStefansson's worry about maintenance. There is no way we can automatize detection of errors for this code, and in practice this will soon lead to more than a slight maintenance overhead.

My suggestion is that if this is important, and in this case it may very well be, it should enter a tutorial, which, btw, need not be very long.

Reply:
Ok. I will write the tutorials when time permits.